falcon/api.py

```diff
@@ -135,13 +136,12 @@ def process_response(req, resp)
     _STREAM_BLOCK_SIZE = 8 * 1024  # 8 KiB
 
     __slots__ = ('_after', '_before', '_request_type', '_response_type',
-                 '_error_handlers', '_media_type', '_routes', '_sinks',
+                 '_error_handlers', '_media_type', '_router', '_sinks',
                  '_serialize_error', 'req_options', '_middleware')
 
     def __init__(self, media_type=DEFAULT_MEDIA_TYPE, before=None, after=None,
                  request_type=Request, response_type=Response,
-                 middleware=None):
-        self._routes = []
+                 middleware=None, router=None):
```

Comment:
Could you add a comment about the `router` arg to the class docstring? Also, it would be great to add an RST page under `doc/api` describing the expected interface in more detail.